src/foam/core/Validation.js

```diff
@@ -115,12 +115,18 @@ foam.CLASS({
             .map(vp => vp.args)
             .flat());
           return [args, function() {
+            var err;
+
             for ( var i = 0 ; i < validationPredicates.length ; i++ ) {
               var vp = validationPredicates[i];
               var self = this;
-              if ( vp.jsFunc.call(this) ) return vp.jsErr.call(self, self);
+              if ( vp.jsFunc.call(this) ) {
+                err = vp.jsFunc.call(self, self);
+              }
+              if ( err )
+                break;
             }
-            return null;
```

Comment:
If you just removed this line and kept everything else the same, then it would behave the same as your changes.

Reply:
fixed. my bad. I should've paid more attention to new changes when merged master